### PR TITLE
Fix bug leading to growing of bin edges by an epsilon in bin/hist/rebin

### LINKED ISSUE #2837: Wrong edge setup in `bin` and `hist` with existing bin-edges
Consider:
```python
import numpy as np
import scipp as sc

size = 100000
var = sc.ones(dims=('row',), shape=(size,))
x = sc.array(dims=['row'], values=86400 * np.random.rand(size))
x.values[0] = 0.0
x.values[x.values>82800] = 82800
da = sc.DataArray(var, coords={'x':x})
edges = sc.linspace(dim='x', start=0, stop=24 * 60 * 60, num=6)
binned = da.bin(x=edges)
binned.hist(x=24).plot()
```
In the final `hist` call, the edges are setup with an upper end equal to the *current* end edge *plus epsilon*. We should not do that, unless the `max` is computed from an event coord. Above this means that events that should end up in the last bin, end up in the second to last.

~The potentially bigger problem is that I see a case related to this where the events are then mapped into *bin 0* (not the previous bins). I have not managed to reproduce this is a small example yet.~ This is an artifact of the discrete param values I used -- all the data got shifted left by one bin, making the first bin a combination of 2 bins.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #2837.

### CODE REVIEW

File: src/scipp/core/binning.py [hunk 1]
```diff
@@ -167,20 +167,20 @@ def _parse_coords_arg(x, name, arg):
     if isinstance(arg, Variable) and name in arg.dims:
         return arg
     coord = _get_coord(x, name)
+    start = coord.min()
+    if not isinstance(x, Variable) and (name in x.meta) and x.meta.is_edges(name, name):
+        stop = coord.max()  # existing bin-edges, do not extend
+    else:
+        stop = _upper_bound(coord)
```
Comment (jl-wynen) [inline src/scipp/core/binning.py:174]: The type hint in `_upper_bound` isn't correct here. If `x` is a data array but does not have a bin edge coord in `dim=name`, we still hit this branch and call `_upper_bound`.

File: src/scipp/core/binning.py [hunk 2]
```diff
@@ -167,20 +167,20 @@ def _parse_coords_arg(x, name, arg):
     if isinstance(arg, Variable) and name in arg.dims:
         return arg
     coord = _get_coord(x, name)
+    start = coord.min()
+    if not isinstance(x, Variable) and (name in x.meta) and x.meta.is_edges(name, name):
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/scipp/core/binning.py:171]: Correct.

File: src/scipp/core/binning.py (hunk 1 above)
Comment (jl-wynen) [reply src/scipp/core/binning.py:174]: Never mind, misread the code

File: docs/about/release-notes.rst [hunk 3]
```diff
@@ -54,7 +54,7 @@ Breaking changes
 Bugfixes
 ~~~~~~~~
 
-* Fix an issue in the setup of automatic bin edges when giving a bin count to :py:func:`scipp.bin`, :py:func:`scipp.hist`, and :py:func:`scipp.rebin`. This lead to a miniscule shift in the resulting edges, which is irrelevant in most cases `#2838 <https://github.com/scipp/scipp/pull/2838>`_.
+* Fix an issue in the setup of automatic bin edges when giving a bin count to :py:func:`scipp.bin`, :py:func:`scipp.hist`, and :py:func:`scipp.rebin`. This lead to a miniscule shift in the resulting edges, which is irrelevant in most cases, but results in events at the lower bin bound to be placed in a correct but lower bin than expected `#2838 <https://github.com/scipp/scipp/pull/2838>`_.
```
Comment (jl-wynen) [inline docs/about/release-notes.rst:57]:
```suggestion
* Fix an issue in the setup of automatic bin edges when giving a bin count to :py:func:`scipp.bin`, :py:func:`scipp.hist`, and :py:func:`scipp.rebin`. This lead to a miniscule shift in the resulting edges, which is irrelevant in most cases, but results in events at the lower bin bound to be placed in a lower bin than expected `#2838 <https://github.com/scipp/scipp/pull/2838>`_.
```

**Review by jl-wynen — APPROVED** [review]

File: docs/about/release-notes.rst (hunk 3 above)
Comment (SimonHeybrock) [reply docs/about/release-notes.rst:57]: Why? It is not in the wrong bin, the edges are wrong.

Comment (jl-wynen) [reply docs/about/release-notes.rst:57]: hm, subtle difference, but ok. Go ahead with merging

**Pull request merged**